public/app/features/dashboard/state/initDashboard.ts

```diff
@@ -106,6 +107,47 @@ async function fetchDashboard(
   }
 }
 
+const trackQueriesOnInitDashboard = (
```

Comment:
**Nit:** maybe I would call this something like `trackQueriesForPanels()`

My reasoning is that the function itself has no knowledge of _when_ it is being called, it takes the queries by datasource from the passed in panels and calls the `trackQueries()` method on the associated datasources. (I think a function named something like `onInitDashboard...` could make sense if it was instantiating different tasks that has to happen on init)